services/ens/api.go

```diff
@@ -399,6 +420,7 @@ func (api *API) Register(ctx context.Context, chainID uint64, txArgs transaction
 	}
 
 	go api.rpcFiltersSrvc.TriggerTransactionSentToUpstreamEvent(types.Hash(tx.Hash()))
+	api.Add(ctx, chainID, fullDomainName(username))
```

Comment:
Is there no other function where we attach the username? I think setPubKey too

Reply:
Ok, I see it now. Didn't find it, because we weren't saving it before 🙂 
Nice catch, thank you!